### ISSUE TITLE
same table name in "memory" and attached database

### ISSUE THREAD

**Issue by antonylebechec** [duckdb/duckdb#6707]
### What happens?

Dear duckdb team,

First, thanks a lot for your developments!!!

in a database connexion (python), with a table named 'variants', I try to attach another database with a table with the same name 'variants', but with different columns (same columns plus one more named 'INFO_nci60').

Here is the error message:
```
duckdb.BinderException: Binder Error: Table "variants" does not have a column named "INFO_nci60"
```

### To Reproduce

Here are my code steps (briefly):
- conn.connect(':memory:')
- conn.execute('CREATE TABLE variants ("ID" INTEGER, "COL1" VARCHAR)')
- conn.execute('INSERT INTO variants VALUES (1, 'one')
- conn.execute("ATTACH DATABASE 'my.duckdb' AS external_db")
with 'my.duckdb' a database with a table named 'variants' with columns: ("ID" INTEGER, "COL1" VARCHAR, "COL2" VARCHAR)

### OS:

almalinux 8.7 (under docker macOS)

### DuckDB Version:

0.7.1

### DuckDB Client:

Python 3.9

### Full Name:

Antony Le Bechec

### Affiliation:

Strasbourg University Hospital (France)

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by paul-iqmo** [comment]
Tried to follow steps, but didn't reproduce error:

dbfile = "file5.db"
import duckdb
con = duckdb.connect(dbfile)
con.execute('CREATE TABLE variants ("ID" INTEGER, "COL1" VARCHAR, "COL2" VARCHAR)')
con.close()

con2 = duckdb.connect(':memory:')
con2.execute('CREATE TABLE variants ("ID" INTEGER, "COL1" VARCHAR)')
con2.execute("INSERT INTO variants VALUES (1, 'one')")
con2.execute(f"ATTACH DATABASE '{dbfile}' AS external_db").df()

**Comment by antonylebechec** [comment]
Thank for your reply!

I think I have a clue.

My external database (file5.db in your exemple) has an index on column COL2. I guess this index is apply to memory.variants when file5.db is attached.

Here is the code to reproduce:
```
import duckdb

dbfile = "file5.db"
con = duckdb.connect(dbfile)
con.execute('CREATE TABLE IF NOT EXISTS variants ("ID" INTEGER, "COL1" VARCHAR, "COL2" VARCHAR)')
con.execute('CREATE INDEX IF NOT EXISTS "idx_COL2" ON variants ("COL2")')
con.close()

con2 = duckdb.connect(':memory:')
con2.execute('CREATE TABLE variants ("ID" INTEGER, "COL1" VARCHAR)')
con2.execute("INSERT INTO variants VALUES (1, 'one')")
con2.execute(f"ATTACH DATABASE '{dbfile}' AS external_db").df()
```

and the error message:

```
Traceback (most recent call last):
  File "/mnt/BIOINFO/git/HOWARD/tests/test.external_db.py", line 12, in <module>
    con2.execute(f"ATTACH DATABASE '{dbfile}' AS external_db").df()
duckdb.BinderException: Binder Error: Table "variants" does not have a column named "COL2"
```

**Comment by paul-iqmo** [comment]
This seems related to https://github.com/duckdb/duckdb/issues/6666? 

In 6666, the attached index wasn't using the attached database tables. 

In this one, the attached index is using the current database.

**Comment by antonylebechec** [comment]
It's definitely the same issue

**Comment by Mytherin** [comment]
Should be fixed by #7481
